### PR TITLE
Mmann1123 ml unsupervised

### PR DESCRIPTION
**Pull request by mmann1123**
## Changes
- Added ability to use unsupervised classifiers like kmeans
- Reordered parameters in fit 
- Removed cross_validation 
     - open to discussion on this, but hoping to rework flow so extract > hyperparameter_tuning > cross_validate > best_model > predict
- added docs for unsupervised

## To do
- Work out tuning and validation workflow from extract
- Figure  out sklearn like validators like `# from sklearn.utils.validation import check_X_y, check_is_fitted, check_array` for sklearn_xarray

### CODE REVIEW

**Comment by mmann1123** [comment]
@jgrss  Made some good additions here. Now we can use unsupervised classifiers!

File: doc/source/ml.rst [hunk 1]
```diff
@@ -95,6 +95,34 @@ Train a classifier and predict
     # Fit the classifier
     with gw.config.update(ref_res=100):
         with gw.open(l8_224078_20200518, chunks=128) as src:
-            X, clf = fit(src, labels, pl, col="lc")
+            X, clf = fit(src, pl, labels, col="lc")
             y = predict(X, clf)
             print(y)
+
+
+Train a unsupervised classifier and predict
```
Comment (jgrss) [inline doc/source/ml.rst:103]: Minor -> an unsupervised

File: geowombat/ml/classifiers.py [hunk 2]
```diff
@@ -5,12 +5,14 @@
 
 import xarray as xr
 from sklearn.pipeline import Pipeline
-from sklearn.model_selection import GridSearchCV, GroupShuffleSplit
 from sklearn.preprocessing import LabelEncoder
 from sklearn_xarray import wrap, Target
```
Comment (jgrss) [inline geowombat/ml/classifiers.py:9]: Is `sklearn_xarray` still being developed?

File: geowombat/ml/classifiers.py [hunk 3]
```diff
@@ -73,12 +64,16 @@ def _prepare_labels(data, labels, col, targ_name):
...
+    def _stack_it(data):
         # TODO: where are we importing Stackerizer from?
-        X = Stackerizer(stack_dims=("y", "x", "time"), direction="stack").fit_transform(
-            data
-        )
+        return Stackerizer(
+            stack_dims=("y", "x", "time"), direction="stack"
+        ).fit_transform(data)
+
+    # @staticmethod
```
Comment (jgrss) [inline geowombat/ml/classifiers.py:73]:
```python
# @staticmethod
```
```suggestion
```

File: geowombat/ml/classifiers.py [hunk 4]
```diff
@@ -189,34 +184,30 @@ class Classifiers(ClassifiersMixin):
...
         Returns:
             y (sklearn_xarray):  Target Data
             Xna (xarray.DataArray): Reshaped feature data with NAs removed
             y, (sklearn pipeline): Fitted pipeline object
 
-
-
         Example:
             >>> import geowombat as gw
             >>> from geowombat.data import l8_224078_20200518, l8_224078_20200518_polygons
```
Comment (jgrss) [inline geowombat/ml/classifiers.py:213]:
```suggestion
            >>> from geowombat.data import l8_224078_20200518, l8_224078_20200518_polygons
            >>> # Add imports for other modules in example
            >>> ... import Pipeline
```

File: geowombat/ml/classifiers.py [hunk 5]
```diff
@@ -234,30 +225,50 @@ def fit(
...
+                coord=targ_name,
+                transform_func=LabelEncoder().fit_transform,
+                dim=targ_dim_name,
+            )(Xna)
+
+            # TO DO: Validation checks
+            # Xna, y = check_X_y(Xna, y)
+
+            clf.fit(Xna, y)
+            self.classes_ = np.unique(y)
```
Comment (jgrss) [inline geowombat/ml/classifiers.py:271]: Do we have a default for `classes_`? Better yet, make it a `property` with a `setter`?

**Review by jgrss — APPROVED** [review]
Nice additions

File: geowombat/ml/classifiers.py (hunk 2 above)
Comment (mmann1123) [reply geowombat/ml/classifiers.py:9]: Good question. Doesn't seem to be actively developed.  Should we try to bring some of those functions into geowombat?

**Review by mmann1123 — COMMENTED** [review]
@jgrss Merging

**Pull request merged**